File: src/main/java/net/lomeli/trophyslots/core/network/MessageSlotsClient.java
package net.lomeli.trophyslots.core.network;

import io.netty.buffer.ByteBuf;

import cpw.mods.fml.common.network.simpleimpl.IMessage;
import cpw.mods.fml.common.network.simpleimpl.IMessageHandler;
import cpw.mods.fml.common.network.simpleimpl.MessageContext;

import net.lomeli.trophyslots.TrophySlots;

public class MessageSlotsClient implements IMessage, IMessageHandler<MessageSlotsClient, IMessage> {
    private int slots;
    private boolean setOrder, reverse;

    public MessageSlotsClient() {
    }

    public MessageSlotsClient(int slots) {
        this.slots = slots;
        this.setOrder = false;
    }

    public MessageSlotsClient(int slots, boolean reverse) {
        this.slots = slots;
        this.setOrder = true;
        this.reverse = reverse;
    }

    @Override
    public void fromBytes(ByteBuf buf) {
        this.slots = buf.readInt();
        this.setOrder = buf.readBoolean();
        if (this.setOrder)
            this.reverse = buf.readBoolean();
    }

    @Override
    public void toBytes(ByteBuf buf) {
        buf.writeInt(this.slots);
        buf.writeBoolean(this.setOrder);
        if (this.setOrder)
            buf.writeBoolean(this.reverse);
    }

    @Override
    public IMessage onMessage(MessageSlotsClient message, MessageContext ctx) {
        TrophySlots.proxy.setSlotsUnlocked(message.slots);
        if (message.setOrder)
            TrophySlots.proxy.setReverse(message.reverse);
        return null;
    }
}

File: src/main/kotlin/net/lomeli/trophyslots/core/network/MessageSlotsClient.kt
package net.lomeli.trophyslots.core.network

import io.netty.buffer.ByteBuf
import net.lomeli.trophyslots.TrophySlots
import net.minecraftforge.fml.common.network.simpleimpl.IMessage
import net.minecraftforge.fml.common.network.simpleimpl.IMessageHandler
import net.minecraftforge.fml.common.network.simpleimpl.MessageContext

public class MessageSlotsClient : IMessage, IMessageHandler<MessageSlotsClient, IMessage> {
    private var slots = 0
    private var startingSlots = 0
    private var setOrder = false
    private var reverse = false

    constructor(){
    }

    constructor(slots: Int) {
        this.slots = slots;
        setOrder = false
    }

    constructor(slots: Int, reverse: Boolean, startingSlots: Int) {
        this.slots = slots
        this.setOrder = true
        this.reverse = reverse
        this.startingSlots = startingSlots
    }

    override fun fromBytes(buf: ByteBuf) {
        this.slots = buf.readInt()
        this.setOrder = buf.readBoolean()
        if (this.setOrder) {
            this.reverse = buf.readBoolean()
            this.startingSlots = buf.readInt()
        }
    }

    override fun toBytes(buf: ByteBuf) {
        buf.writeInt(this.slots)
        buf.writeBoolean(this.setOrder)
        if (this.setOrder) {
            buf.writeBoolean(this.reverse)
            buf.writeInt(this.startingSlots)
        }
    }

    override fun onMessage(message: MessageSlotsClient?, ctx: MessageContext?): IMessage? {
        if (message != null) {
            TrophySlots.proxy.setSlotsUnlocked(message.slots)
            if (message.setOrder) {
                TrophySlots.proxy.setReverse(message.reverse)
                TrophySlots.proxy.startingSlots = message.startingSlots
            }
        }
        return null
    }
}